### ISSUE TITLE
why are my bots stopping / or maybe i am doing it wrong

### ISSUE THREAD

**Issue by mathse** [shirosaidev/diskover#48]
my basic scan setup looks like the following

name=$(generate a name)
./diskover-bot-launcher.sh -w 128
./diskover.py -i $name -d /mnt/point $OPTIONS
./diskover.py -i $name -D

while scanning I am looking at the redis instance

[root@sl-it-p-crwlr1 2019-01]# rqinfo --url redis://localhost:6379
diskover_crawl |█████████████████ 174
1 queues, 174 jobs total
0 workers, 1 queues
Updated: 2019-01-06 11:26:37.474399

I am experiancing that diskover is running out of workers :-/
all the other 128 workers did not log anything to the log files
at the same time diskover is still able to work jobs from the queue ... without any workers?

**Comment by shirosaidev** [comment]
do any workers start?
kill all workers and then run 1 worker in foreground and then after start crawl
`python diskover_worker_bot.py -l DEBUG`
What's in the log output for the worker, is it processing jobs?

**Comment by mathse** [comment]
yes its processing jobs - looks like it does not register to the redis queue probperly

**Comment by mathse** [comment]
this is what a bots last output is
`10:55:59 Sent heartbeat to prevent worker timeout. Next one should arrive within 420 seconds.
10:55:59 Sent heartbeat to prevent worker timeout. Next one should arrive within 35 seconds.
10:56:16 diskover_crawl: Job OK (e2e7d9d2-2f02-4da8-92d7-aeff8417b262)
10:56:16 Result is kept for 500 seconds
10:56:16 Sent heartbeat to prevent worker timeout. Next one should arrive within 420 seconds.
10:56:16 Sent heartbeat to prevent worker timeout. Next one should arrive within 420 seconds.
10:56:16 *** Listening on diskover,diskover_crawl,diskover_calcdir...
10:56:16 Sent heartbeat to prevent worker timeout. Next one should arrive within 420 seconds.`

**Comment by shirosaidev** [comment]
it doesn't register even when the bot is starting up? If this is the cace, stop all bots, check there is not bot pids running, then empty queues and then flush redis using `redis-cli flushall`. Then start up one bot and see if it registers in rq/redis. Other people are having similiar issue with bots disappearing during crawl...

**Comment by mathse** [comment]
when i start the bots they register just fine to redis ... once they start working the slowly decrease

**Comment by shirosaidev** [comment]
I've pushed some code changes to rc28 that hopefully addresses this, see changelog and also check that you have set these optimization settings for redis server
https://github.com/shirosaidev/diskover/wiki/Redis
